test/e2e/fixture/zomes/foo/src/lib.rs

```diff
@@ -20,15 +20,25 @@ impl From<&str> for TestString {
 #[hdk_extern]
 fn init(_: ()) -> ExternResult<InitCallbackResult> {
     // grant unrestricted access to accept_cap_claim so other agents can send us claims
-    let mut functions: GrantedFunctions = HashSet::new();
-    functions.insert((zome_info!()?.zome_name, "foo".into()));
-    // functions.insert((zome_info!()?.zome_name, "needs_cap_claim".into()));
+    let mut foo_functions: GrantedFunctions = HashSet::new();
+    foo_functions.insert((zome_info!()?.zome_name, "foo".into()));
     create_cap_grant!(
         CapGrantEntry {
             tag: "".into(),
             // empty access converts to unrestricted
             access: ().into(),
-            functions,
+            functions: foo_functions,
+        }
+    )?;
+
+    let mut emitter_functions: GrantedFunctions = HashSet::new();
+    emitter_functions.insert((zome_info!()?.zome_name, "emitter".into()));
+    create_cap_grant!(
+        CapGrantEntry {
+            tag: "".into(),
+            // empty access converts to unrestricted
+            access: ().into(),
+            functions: emitter_functions,
         }
```

Comment:
I think this is a problem, even if this works now, because the intended behavior is that there is only ever one unrestricted capability grant. The expected behavior is that the latter grant would overwrite the prior grant. I think there should just be a single grant with both functions.

Reply:
Oo... i tried that, and it resulted in errors... i will have to log the error i saw because it was in holochain

Reply:
@maackle 
if I just use one unrestricted cap grant for the two functions the tests throw this...
```rust
    let mut foo_functions: GrantedFunctions = HashSet::new();
    foo_functions.insert((zome_info!()?.zome_name, "foo".into()));
    foo_functions.insert((zome_info!()?.zome_name, "emitter".into()));
    create_cap_grant!(
        CapGrantEntry {
            tag: "".into(),
            // empty access converts to unrestricted
            access: ().into(),
            functions: foo_functions,
        }
    )?;
```

```
conductor>  Holochain detected database corruption.
        Corrupt module: CasBufUsedSync::get
        Expected hash: [3, 199, 87, 4, 16, 146, 251, 131, 169, 175, 186, 194, 194, 149, 231, 252, 135, 95, 212, 71, 189, 96, 153, 176, 114, 223, 44, 34, 241, 116, 214, 135, 146, 205, 224, 56]
        Actual hash: [212, 30, 52, 37, 86, 225, 147, 222, 39, 211, 204, 55, 169, 215, 90, 79, 83, 95, 252, 113, 240, 75, 156, 217, 111, 106, 226, 163, 27, 109, 129, 140, 2, 1, 201, 58]
        Content: CapGrant(ZomeCallCapGrant { tag: "", access: Unrestricted, functions: {(ZomeName("foo"), FunctionName("foo")), (ZomeName("foo"), FunctionName("emitter"))} })
        We are shutting down as a precaution to prevent further corruption.
        FATAL PANIC PanicInfo {
            payload: Any,
            message: Some(
                Holochain detected database corruption.
                Corrupt module: CasBufUsedSync::get
                Expected hash: [3, 199, 87, 4, 16, 146, 251, 131, 169, 175, 186, 194, 194, 149, 231, 252, 135, 95, 212, 71, 189, 96, 153, 176, 114, 223, 44, 34, 241, 116, 214, 135, 146, 205, 224, 56]
                Actual hash: [212, 30, 52, 37, 86, 225, 147, 222, 39, 211, 204, 55, 169, 215, 90, 79, 83, 95, 252, 113, 240, 75, 156, 217, 111, 106, 226, 163, 27, 109, 129, 140, 2, 1, 201, 58]
                Content: CapGrant(ZomeCallCapGrant { tag: "", access: Unrestricted, functions: {(ZomeName("foo"), FunctionName("foo")), (ZomeName("foo"), FunctionName("emitter"))} })
                We are shutting down as a precaution to prevent further corruption.,
            ),
            location: Location {
                file
        conductor>  : "/Users/connor/.rustup/toolchains/stable-2020-08-03-x86_64-apple-darwin/lib/rustlib/src/rust/src/libstd/macros.rs",
                line: 16,
                col: 9,
            },
        }
        thread 'holochain-tokio-thread' panicked at 'Holochain detected database corruption.
        Corrupt module: CasBufUsedSync::get
        Expected hash: [3, 199, 87, 4, 16, 146, 251, 131, 169, 175, 186, 194, 194, 149, 231, 252, 135, 95, 212, 71, 189, 96, 153, 176, 114, 223, 44, 34, 241, 116, 214, 135, 146, 205, 224, 56]
        Actual hash: [212, 30, 52, 37, 86, 225, 147, 222, 39, 211, 204, 55, 169, 215, 90, 79, 83, 95, 252, 113, 240, 75, 156, 217, 111, 106, 226, 163, 27, 109, 129, 140, 2, 1, 201, 58]
        Content: CapGrant(ZomeCallCapGrant { tag: "", access: Unrestricted, functions: {(ZomeName("foo"), FunctionName("foo")), (ZomeName("foo"), FunctionName("emitter"))} })
        We are shutting down as a precaution to prevent further corruption.', /Users/connor/.rustup/toolchains/stable-2020-08-03-x86_64-apple-darwin/lib/rustlib/src/rust/src/libstd/macros.rs:16:9
        stack backtrace:
        conductor>     0: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
        conductor>   1: core::fmt::write
           2: std::io::Write::write_fmt
           3: std::panicking::default_hook::{{closure}}
           4: std::panicking::default_hook
           5: <holochain_state::env::ENVIRONMENTS as core::ops::deref::Deref>::deref::__static_ref_initialize::{{closure}}
           6: std::panicking::rust_panic_with_hook
           7: rust_begin_unwind
           8: std::panicking::begin_panic_fmt
           9: holochain_state::buffer::cas::buf_sync::CasBufUsedSync<C,P>::deserialize_and_hash
          10: holochain::core::state::source_chain::
        conductor>  SourceChain::valid_cap_grant::{{closure}}
          11: holochain::core::state::source_chain::SourceChain::valid_cap_grant
          12: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
          13: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::
        conductor>  poll
          14: std::thread::local::LocalKey<T>::with
          15: futures_executor::local_pool::block_on
          16: tokio::runtime::enter::exit
          17: tokio::runtime::thread_pool::worker::
        conductor>  block_in_place
          18: holochain::core::ribosome::ZomeCallInvocation::is_authorized
          19: <holochain::core::ribosome::wasm_ribosome::WasmRibosome as holochain::core::ribosome::RibosomeT>::call_zome_function
        conductor>    20: holochain::core::workflow::call_zome_workflow::call_zome_workflow_inner::{{closure}}
          21: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
          22: <core::future::from_generator::
        conductor>  GenFuture<T> as core::future::future::Future>::poll
          23: <core::future::from_generator::GenFuture<T> as core
        conductor>  ::future::future::Future>::poll
          24: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
        conductor>  25: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
          26: <core::future::from_generator::GenFuture<T> as core::future::future::
        conductor>  Future>::poll
          27: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
          28: <core::future::from_generator::GenFuture
        conductor>  <T> as core::future::future::Future>::poll
          29: tokio::runtime::task::core::Core<T,S>::poll
          30: <
        conductor>  std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
          31: tokio::runtime::task::raw::poll
          32: std::thread::local
        conductor>  ::LocalKey<T>::with
          33: tokio::runtime::thread_pool::worker::Context::run_task
          34: tokio::runtime::thread_pool::worker::Context::run
          35: tokio::macros
        conductor>  ::scoped_tls::ScopedKey<T>::set
          36: tokio::runtime::thread_pool::worker::run
          37: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
          38: <
        conductor>  std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
          39: tokio::runtime::task::harness::Harness<T,S
        conductor>  >::poll
          40: tokio::runtime::blocking::pool::Inner::run
          41: tokio::runtime::context::enter
        note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

Reply:
the same bug occured with me as well in [#418](https://github.com/holochain/holochain/issues/418)

Reply:
oh! thank you for noting @tatssato